packet/Device.py

```diff
@@ -65,6 +65,18 @@ def update(self):
     def delete(self):
         return self.manager.call_api("devices/%s" % self.id, type="DELETE")
 
+    def reinstall(self, operating_system=None, ipxe_script_url=None):
+        params = {"type": "reinstall"}
+        if operating_system is not None:
+            params["operating_system"] = operating_system
+        if ipxe_script_url is not None:
```

Comment:
Was going to mention about erroring if `operating_system` is set to !`custom_ipxe` but `ipxe_script_url` is passed in but then noticed we do this override in the regular provision flow so nm.

Reply:
Yeah, the other thing is there is a valid UUID name for custom_ipxe too ... and then just "meh"'d my way to here :P